scipy/spatial/tests/test_hausdorff.py

```diff
@@ -0,0 +1,130 @@
+        # Ensure that the algorithm for directed_hausdorff gives the
+        # same result as the simple / brute force approach in the
+        # reverse direction.
+        actual = directed_hausdorff(self.path_2, self.path_1)[0]
+        # brute force over columns:
+        expected = max(np.amin(distance.cdist(self.path_1, self.path_2), 
+                               axis=0))
+        assert_almost_equal(actual, expected, decimal=9)
+
+    def test_degenerate_case(self):
+        # The directed Hausdorff distance must be zero if both input
+        # data arrays match.
+        actual = directed_hausdorff(self.path_1, self.path_1)[0]
+        assert_almost_equal(actual, 0.0, decimal=9)
+
+    def test_2d_data_forward(self):
+        # Ensure that 2D data is handled properly for a simple case
+        # relative to brute force approach.
+        actual = directed_hausdorff(self.path_1[..., :2],
+                                    self.path_2[..., :2])[0]
+        expected = max(np.amin(distance.cdist(self.path_1[..., :2],
+                                              self.path_2[..., :2]),
+                               axis=1))
+        assert_almost_equal(actual, expected, decimal=9)
+
+    def test_4d_data_reverse(self):
+        # Ensure that 4D data is handled properly for a simple case
+        # relative to brute force approach.
+        actual = directed_hausdorff(self.path_2_4d, self.path_1_4d)[0]
+        # brute force over columns:
+        expected = max(np.amin(distance.cdist(self.path_1_4d, self.path_2_4d), 
+                               axis=0))
+        assert_almost_equal(actual, expected, decimal=9)
+
+    def test_indices(self):
+        # Ensure that correct point indices are returned -- they should
+        # correspond to the Hausdorff pair
+        path_simple_1 = np.array([[-1,-12],[0,0], [1,1], [3,7], [1,2]])
+        path_simple_2 = np.array([[0,0], [1,1], [4,100], [10,9]])
+        actual = directed_hausdorff(path_simple_2, path_simple_1)[1:]
+        expected = (2, 3)
+        assert_array_equal(actual, expected)
+
+    def test_random_state(self):
+        # ensure that the global random state is not modified because
+        # the directed Hausdorff algorithm uses randomization
+        rs = check_random_state(None)
+        old_global_state = rs.get_state()
+        directed_hausdorff(self.path_1, self.path_2)
+        rs2 = check_random_state(None)
+        new_global_state = rs2.get_state()
+        assert_equal(new_global_state, old_global_state)
+
+    def test_random_state_None(self):
+        # check that a seed value of None does not alter global random
+        # state
+        rs = check_random_state(None)
+        old_global_state = rs.get_state()
+        directed_hausdorff(self.path_1, self.path_2, None)
+        rs2 = check_random_state(None)
+        new_global_state = rs2.get_state()
+        assert_equal(new_global_state, old_global_state)
+
+    def test_random_state_int(self):
```

Comment:
there's a bit of code duplication going on here; better to combine with the test above (just put it in ``for seed in [None, 27870671]:``